### PR TITLE
TST: Timestamp and Timeseries tests reorg (gh14854)

### PR DESCRIPTION
**Pull request by TrigonaMinima**
- [x] Partially completes #14854 
 - [x] Tests reorganized (```test_timeseries.py``` + ```test_tslib.py``` -> ```test_timestamp.py```)
 - [x] Passes ``git diff upstream/master | flake8 --diff``

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. the remainder of ``tseries/test_timeseries.py`` can prob go in ``tests/series/test_timestamp.py`` (or can move the obvious stuff). Then can look and see what remains.

thanks!

**Comment by TrigonaMinima** [comment]
Wouldn't the remaining tests of ```tseries/test_timeseries.py``` go in ```tests/series/test_timeseries.py``` instead of ```tests/series/test_timestamp.py```? You have also mentioned the same in the issue ( #14854 3rd point).

**Comment by jreback** [comment]
@TrigonaMinima yes, sorry, that's what I meant!

of course if you find say a period test or something, move to the appropriate place.

**Comment by jreback** [comment]
ok a lot of the move into tests/series/test_timestamp.py was composed of things (that can be split out)

- fillna (on series) can go to test_missing
- move pure date_range tests an go to indexes/datetimes/test_date_range (new file)
- maybe some pure Timestamp things to scalar/test_timestamp

**Comment by TrigonaMinima** [comment]
You meant ```tests/series/time_series.py```, right? :smile: 

It's still large though.

**Comment by jreback** [comment]
@TrigonaMinima yeah. Need to separate pieces out as much as possible (though still may end up large). ok for now. I'll take a look and move more.

File: pandas/tests/series/test_missing.py [hunk 1]
```diff
@@ -530,6 +535,79 @@ def test_fill_value_when_combine_const(self):
...
+
+        ss = s[-2:].reindex(index).to_sparse()
+        result = ss.fillna(method='backfill', limit=5)
+        expected = ss.fillna(method='backfill')
+        expected = expected.to_dense()
+        expected[:3] = np.nan
+        expected = expected.to_sparse()
+        assert_series_equal(result, expected)
+
+    def test_sparse_series_pad_backfill_limit(self):
```
Comment (jreback) [inline pandas/tests/series/test_missing.py]: forgot to mention, these sparse routines can go in 

pandas/sparse/tests/test_series.py

File: pandas/tests/series/test_missing.py [hunk 2]
```diff
@@ -530,6 +535,79 @@ def test_fill_value_when_combine_const(self):
...
+        assert_series_equal(result, expected)
+
+        result = s[-2:].reindex(index, method='backfill', limit=5)
+        expected = s[-2:].reindex(index).fillna(method='backfill')
+        expected = expected.to_dense()
+        expected[:3] = np.nan
+        expected = expected.to_sparse()
+        assert_series_equal(result, expected)
+
+    def test_series_pad_backfill_limit(self):
```
Comment (jreback) [inline pandas/tests/series/test_missing.py]: anything backfill or interpolate can also do in pandas/tests/series/test_missing

File: pandas/tests/series/test_missing.py [hunk 3]
```diff
@@ -530,6 +535,79 @@ def test_fill_value_when_combine_const(self):
...
+        assert_series_equal(result, expected)
+
+        result = s[-2:].reindex(index, method='backfill', limit=5)
+
+        expected = s[-2:].reindex(index).fillna(method='backfill')
+        expected[:3] = np.nan
+        assert_series_equal(result, expected)
+
 
 class TestSeriesInterpolateData(TestData, tm.TestCase):
```
Comment (jreback) [inline pandas/tests/series/test_missing.py]: in fact this whole class can be moved

File: pandas/tests/series/test_missing.py (hunk 2 above)
Comment (jreback) [inline pandas/tests/series/test_missing.py]: sorry, I realized you already moved this, nvm!

**Comment by jreback** [comment]
@TrigonaMinima just ping on green. I'll do a follow on this stuff to fix a couple of little points.

If you can continue on to other things would be great!

File: pandas/tests/series/test_missing.py (hunk 2 above)
Comment (TrigonaMinima) [inline pandas/tests/series/test_missing.py]: They are in test_missing only. Though there are some frame backfill routines which are in ```test_timeseries.py```. Should they be moved here as well?

File: pandas/tests/series/test_missing.py (hunk 3 above)
Comment (TrigonaMinima) [inline pandas/tests/series/test_missing.py]: already in test_missing :slightly_smiling_face:

**Comment by jreback** [comment]
@TrigonaMinima don't push anymore to this. I have already taken your commit (and going to merge it soon), along with a bunch of other changes.

**Comment by TrigonaMinima** [comment]
Sure. When I pushed it there wasn't anything. Sorry for that.

**Comment by jreback** [comment]
@TrigonaMinima no prob. 

https://github.com/pandas-dev/pandas/compare/master...jreback:fix-tests?expand=1

is the diff

**Comment by jreback** [comment]
@TrigonaMinima if you are game, love to move timedelta and period in a similar way (e.g. move the scalar tests and index tests). you can create a ``indexes/timedeltas,period`` sub-dir

e.g. for timedelta separate out the ``to_timedelta`` stuff into a separate file apart from *multiple* files for timedelta index consruction / ops etc.

similar for period.

**Comment by jreback** [comment]
also you can move ``tserites/tests/test_daterange.py`` to ``indexes/datetimes/test_date_range`` (add to same file).

**Comment by jreback** [comment]
and ``tseries/tests/test_tslib.py`` can go to (you will have to move classes / individual test)

- test_timezones.py (only move stuff that is actually doing timezone stuff, not *parsing* timezones, that will be in below)
- indexes/datetimes/test_tools.py (for parsing and tslib.* routines)
- test_period.py 
- a couple of random things at the bottom have to find a good spot.

do this *after* I merge (shortly) and separately from everything else as this will touch lots of things.

**Comment by TrigonaMinima** [comment]
@jreback sure. I just have a couple of questions-

- Why are there tests outside tests directory? What's the difference between these tests in ```pandas/tests``` and in each sub directory under pandas? Is it by design?
- What goes under scalars and what goes under indexes. Some routines seemed to me as if they could go anywhere. So if there is some part of documentation where i can know the difference or you may can me.

Also, just clarifying, I should do timedelta changes here in a different PR right?

**Comment by jreback** [comment]
yes, leave this PR alone. timedelta is a separate PR.

in *theory* everything should be under ``pandas/tests`` (sub-dirs of). but we have historically not done that, so slowly moving things.

so in the scalar dir are ``Timestamp``, ``Timedelta``, ``Period`` specific things, as opposed to ``DatetimeIndex``, ``TimedeltaIndex``, ``PeriodIndex`` (which are the indexes which wrap these scalars).

e.g. a method/property of ``Timestamp`` belongs in scalar, while *using* ``Timestamp`` to say subtract from a ``DatetimeIndex`` belong in indexes.

the tests dir *should* be almost self-explanatory, with it obvious where a new tests should go. there should be 1 and only 1 place.

**Comment by jreback** [comment]
merged!

**Comment by TrigonaMinima** [comment]
Thanks. The explanation made things clearer.

**Pull request closed without merging**